### PR TITLE
Invalid tag modification

### PR DESCRIPTION
**Pull request by bkowshik**
Review invalid tag modifications to primary tags. Ex: `highway` feature has the `highway` tag removed and made a `building`.

---

cc: @amishas157

### CODE REVIEW

File: comparators/invalid-tag-modification.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+function getPrimaryTag(properties) {
+  var primaryTags = [
+    'aerialway', 'aeroway', 'amenity', 'barrier', 'boundary', 'building', 'craft', 'emergency',
+    'geological', 'highway', 'historic', 'landuse', 'leisure', 'man_made', 'military', 'natural',
+    'office', 'places', 'power', 'public_transport', 'railway', 'route', 'shop', 'sport',
+    'tourism', 'waterway'
+  ];
+  for (var key in properties) {
+    if (primaryTags.indexOf(key) !== -1) return key;
```
Comment (amishas157) [inline comparators/invalid-tag-modification.js]: @bkowshik Wondering if a feature has more than one primary tag( We already have a cf for check [bad primary tag combinations](https://github.com/mapbox/osm-compare/blob/master/comparators/invalid-tag-combination.js)). In that case, only the first tag which comes through the loop will be checked for modification. How would it be to return the set of primary tags instead?

Comment (bkowshik) [inline comparators/invalid-tag-modification.js]: @amishas157 I wanted to keep this compare function handle just the case when there is one primary tag. As you said, we have another compare function to flag when there are multiple primary tags in a feature that are unpopular.

**Comment by bkowshik** [comment]
Agree that the name does not say what it is doing straight. But, don't want to make this primary tag specific although it is at present. I wanted to keep it open to handle invalid tag modifications for non-primary tags. An example I can think of is `water`.

---

cc: @amishas157

**Comment by amishas157** [comment]
@bkowshik 
>An example I can think of is water.

By this, do you mean conversion of water in some other non-primary tag?

**Comment by bkowshik** [comment]
Thank you @amishas157 👍 

The current implementation does not handle the case of multiple primary tags in a feature. Will wire up a test fixture and modify the comparator to handle the same.

**Comment by bkowshik** [comment]
Per voice with @amishas157 

- Let's consider the primary tag of a feature as the identify of the feature. Ex: A new feature with the `highway` primary tag can be described as "*This feature is a highway*".
- The identity of the feature **should be retained over multiple versions**. Ex: This highway should remain a `highway` over multiple version and not become a `waterway`.
- This means, once a feature gets a primary tag, it should ideally not loose that primary tag and when a feature looses one, it is invalid.

**Comment by bkowshik** [comment]
One scenario for this compare function is when for Lake Michigan, the primary tag of the feature, `natural=water` was deleted.

- http://osmcha.mapbox.com/44609658/

**Comment by bkowshik** [comment]
This comparator labelled a harmful changeset today.

- http://osmcha.mapbox.com/47487736/

screen shot 2017-04-06 at 5 30 39 pm

But, the *undo* of the changeset was flagged too 😉 and this needs to be fixed.

- https://osmcha.mapbox.com/47497747/

**Comment by amishas157** [comment]
@bkowshik 
A quick thought which I have for this is dealing with these kind of problems at osmcha -django level. For example coupling this cf with new mapper would have flagged http://osmcha.mapbox.com/47487736/ and not https://osmcha.mapbox.com/47497747/

**Comment by amishas157** [comment]
Some results from this combination : 

Harmful and checked : https://osmcha.mapbox.com/?is_suspect=True&is_whitelisted=True&harmful=True&checked=All&reasons%5B%5D=40&reasons%5B%5D=42&all_reason=True

Count : 4

All checked for new Mapper + invalid tag modification: https://osmcha.mapbox.com/?is_suspect=True&is_whitelisted=True&harmful=None&checked=True&reasons%5B%5D=40&reasons%5B%5D=42&all_reason=True

Count : 11

**Pull request merged**